org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectPreferenceChangeListener.java

```diff
@@ -47,11 +47,12 @@ public void preferencesChange(Preferences oldPreferences, Preferences newPrefere
 
 				try {
 					IFolder workspaceLinkFolder = javaProject.getProject().getFolder(ProjectUtils.WORKSPACE_LINK);
-					if (!workspaceLinkFolder.exists()) {
+					IPath rootPath = ProjectUtils.findBelongedWorkspaceRoot(workspaceLinkFolder.getLocation());
```

Comment:
if workspaceLinkFolder does not exist, what is returned by `workspaceLinkFolder.getLocation()`?

Reply:
It depends.

If it's the linked folder of an invisible project, it will return the linked folder path. If it's a visible project, it just appends `_`, for example: `D:/Workspace/spring-petclinic/_`

Why `.exists()` does not work is that, it just check if the handle of the resource exists, not check the real one. Below is part of the document copied from `.exists()`:

`IResource objects are lightweight handle objects used to access resources in the workspace. However, having a handle object does not necessarily mean the workspace really has such a resource.`